Only in fastfetch: fastfetch-2.13.1-1-x86_64.pkg.tar
Binary files fastfetch.ORIG/.git/index and fastfetch/.git/index differ
Only in fastfetch: .patched
Only in fastfetch: pkg
Only in fastfetch/src: build
diff --color -ru fastfetch.ORIG/src/fastfetch-2.18.0/src/detection/host/host_linux.c fastfetch/src/fastfetch-2.18.0/src/detection/host/host_linux.c
--- fastfetch.ORIG/src/fastfetch-2.18.0/src/detection/host/host_linux.c	2024-05-21 14:02:13.474705904 -0700
+++ fastfetch/src/fastfetch-2.18.0/src/detection/host/host_linux.c	2024-05-21 14:19:18.814581438 -0700
@@ -50,9 +50,17 @@
 const char* ffDetectHost(FFHostResult* host)
 {
     ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_family", "/sys/class/dmi/id/product_family", &host->family);
-    if (!ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_name", "/sys/class/dmi/id/product_name", &host->name))
+    if (!ffGetSmbiosValue("/sys/devices/virtual/dmi/id/board_name", "/sys/class/dmi/id/board_name", &host->name))
         getHostProductName(&host->name);
-    ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_version", "/sys/class/dmi/id/product_version", &host->version);
+    if (host->name.length == 0) {
+        if (!ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_name", "/sys/class/dmi/id/product_name", &host->name))
+            getHostProductName(&host->name);
+    }
+
+    ffGetSmbiosValue("/sys/devices/virtual/dmi/id/bios_version", "/sys/class/dmi/id/bios_version", &host->version);
+    if (host->version.length == 0)
+        ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_version", "/sys/class/dmi/id/product_version", &host->version);
+
     ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_sku", "/sys/class/dmi/id/product_sku", &host->sku);
     if (!ffGetSmbiosValue("/sys/devices/virtual/dmi/id/product_serial", "/sys/class/dmi/id/product_serial", &host->serial))
         getHostSerialNumber(&host->serial);
